docs/TheBook/TechnologyTutorial.md

```diff
@@ -0,0 +1,797 @@
+Marcos for properties of the Trigger Object work for both Text and Payload.
+
+## Chapter 8 - Scripting the C-DEngine
+
+The C-DEngine allows to script the creation of Things for headless (without NMI) configuration of your nodes.
+There are two ways of scripting:
+
+* Using .cdeConfig files - this is using the C-DEngine Sensor Pipeline. A detailed description of [this pipeline can be found here](https://github.com/TRUMPF-IoT/cdeDocs/blob/master/docs/Concepts/SensorPipelineModel.md) 
+* Using .cdeScript files - this is using TheThingProvisioning Plugin to send messages to Things at startup
+
+Lets use the cdeConfig File to automatically create the CDMyNetwork "Google DNS" D-Thing:
+
+1) Go to TheThingRegistry and go to the details of your "Google DNS" D-Thing
+2) Click on the blue Details button ![Img6](Images/img6.png)
+
+Now click on the "Export Pipeline Generalized" button.
+
+Your browser will now download a .cdeConfig file with this content:
+
+```JSON
+{
+  "FriendlyName": "Google DNS",
+  "ThingConfigurations": [
+    {
+      "ThingIdentity": {
+        "EngineName": "CDMyNetwork.MyNetworkServices",
+        "DeviceType": "Ping Service",
+        "ID": "327c7a23-54b7-4e29-bd95-fff6fc47ae2a"
+      },
+      "ConfigurationValues": [
+        {
+          "Name": "AutoConnect",
+          "cdeT": 2,
+          "Value": true,
+          "Generalize": false,
+          "IsThingReference": false,
+          "Required": false,
+          "Secure": false
+        },
+        {
+          "Name": "PingDelay",
+          "cdeT": 1,
+          "Value": 3000,
+          "Generalize": false,
+          "IsThingReference": false,
+          "Required": false,
+          "Secure": false
+        },
+        {
+          "Name": "Address",
+          "cdeT": 0,
+          "Value": "8.8.8.8",
+          "Generalize": false,
+          "IsThingReference": false,
+          "Required": false,
+          "Secure": false
+        },
+        {
+          "Name": "PingTimeOut",
+          "cdeT": 1,
+          "Value": 3000,
+          "Generalize": false,
+          "IsThingReference": false,
+          "Required": false,
+          "Secure": false
+        },
+        {
+          "Name": "FailureLimit",
+          "cdeT": 1,
+          "Value": 1,
+          "Generalize": false,
+          "IsThingReference": false,
+          "Required": false,
+          "Secure": false
+        },
+        {
+          "Name": "FriendlyName",
+          "cdeT": 0,
+          "Generalize": true
+        },
+        {
+          "Name": "ID",
+          "cdeT": 0,
+          "Generalize": true
+        }
+      ],
+      "ThingSpecializationParameters": {
+        "FriendlyName": "Google DNS",
+        "ID": "327c7a23-54b7-4e29-bd95-fff6fc47ae2a"
+      }
+    }
+  ]
+}
+```
+
+3) Change the Friendly Name to "Google DNS2" and give the "ID" a different GUID. ***ASK MARKUS: Why is the ID twice here - can you explain?***
```

Comment:
This indicates that you can change the ID using a .cdeanswer file (named the same as the .cdeconfig file), by just removing everything but the ThingSpecializationParameters from the JSON: 

```JSON
{
  "FriendlyName": "My Ping D-Things",
  "ThingConfigurations": [
    {
      "ThingSpecializationParameters": {
        "FriendlyName": "Google DNS",
        "Address": "8.8.8.8",
        "ID": "327c7a23-54b7-4e29-bd95-fff6fc47ae2a"
      }
    },
    {
      "ThingSpecializationParameters": {
        "FriendlyName": "microsoft.com",
        "Address": "www.microsoft.com",
        "ID": "12345"
      }
    }
  ]
}
```

This will create two things using the template information in the .cdeconfig file (which plug-in, ping interval etc.)